src/SW_netCDF_Input.c

```diff
@@ -6299,17 +6538,171 @@ static void read_soil_inputs(
                 addOffset,
                 soilConv[varNum - 1],
                 isSwrcpVar,
-                (!isSwrcpVar) ? 0 : (varNum - swrcpStartInd),
+                (!isSwrcpVar) ? 0 : (varNum - eiv_swrcpMS[0]),
                 loopIter,
                 (!isSwrcpVar) ? doublePtr : swrcpMS[loopIter]
             );
         }
     }
 
-    if (!hasConstSoilLyrs) {
+
+    // Derive missing soil properties from available properties
+    for (slNum = 0; slNum < MAX_LAYERS; slNum++) {
```

Comment:
I suggest we put this chunk into a helper function (e.g., `derive_missing_soils()`) to help simplify `read_soil_inputs()`.